mountpoint-s3/src/fs.rs

```diff
@@ -547,6 +572,28 @@ where
                 file_etag = etag.clone();
                 request.lock().await
             }
+            FileHandleType::Uninitialized { lookup, .. } => {
```

Comment:
Just thinking out loud: is there a way to refactor this so that this logic is shared with the `O_RDONLY` path in `open`, and similar for `O_WRONLY` and the logic in `write`? Would be nice to have those be shared so there's only one place each type of handle can be created.

Might also be nice to find a way to represent `Uninitialized` so that not every code path has to deal with that case. Maybe `typ` can be another enum with `Open(FileHandleType)` and `NotYetOpen(Lookup)` variants or something?

Reply:
Yeah, we can also do it that way. I'll update the handle structure.

Reply:
What do you think if we delay the file handle initialization until the first read or write? That way there should be one place each type can be created, but it will also change current file system behavior.